File: server/app/api/v1/bots/bots.py
"""Coze Bot 完整 CRUD 路由(基于 coze_compat.CozeClient)."""

from fastapi import APIRouter, Depends, Query

from app.schemas.common import error, success
from app.security import require_login
from app.utils.coze_compat import CozeClient

router = APIRouter()


@router.get("/list", summary="Bot 列表")
async def list_bots(
    page: int = Query(1, ge=1),
    page_size: int = Query(20, ge=1, le=100),
    space_id: str = Query("", description="空间 ID,默认使用 settings.COZE_ACCOUNT_ID"),
    user_uuid: str = Depends(require_login),
):
    async with CozeClient() as cli:
        data = await cli.list_bots(space_id=space_id, page=page, size=page_size)
    if data.get("code") != 0:
        return error(data.get("msg", "Coze API error"))
    return success(
        data.get("data", {}).get("bot_list", []),
        total=data.get("data", {}).get("total", 0),
    )


@router.get("/{bot_id}", summary="Bot 详情")
async def get_bot(bot_id: str, user_uuid: str = Depends(require_login)):
    async with CozeClient() as cli:
        data = await cli.get_bot(bot_id)
    if data.get("code") != 0:
        return error(data.get("msg", "Coze API error"))
    return success(data.get("data"))


@router.post("/create", summary="创建 Bot")
async def create_bot(
    name: str = Query(...),
    description: str = Query(""),
    persona: str = Query("", description="Bot 人设描述"),
    user_uuid: str = Depends(require_login),
):
    payload = {
        "space_id": "",  # 由 Coze 端兜底
        "name": name,
        "description": description,
        "persona": {"system_prompt": persona} if persona else {},
    }
    async with CozeClient() as cli:
        data = await cli.create_bot(payload)
    if data.get("code") != 0:
        return error(data.get("msg", "Coze API error"))
    return success(data.get("data"))


@router.post("/update", summary="更新 Bot")
async def update_bot(
    bot_id: str = Query(...),
    name: str = Query(None),
    description: str = Query(None),
    persona: str = Query(None),
    user_uuid: str = Depends(require_login),
):
    payload = {}
    if name is not None:
        payload["name"] = name
    if description is not None:
        payload["description"] = description
    if persona is not None:
        payload["persona"] = {"system_prompt": persona}
    async with CozeClient() as cli:
        data = await cli.update_bot(bot_id, payload)
    if data.get("code") != 0:
        return error(data.get("msg", "Coze API error"))
    return success(data.get("data"))


@router.post("/delete", summary="删除 Bot")
async def delete_bot(bot_id: str = Query(...), user_uuid: str = Depends(require_login)):
    async with CozeClient() as cli:
        data = await cli.delete_bot(bot_id)
    if data.get("code") != 0:
        return error(data.get("msg", "Coze API error"))
    return success({"bot_id": bot_id, "deleted": True})


@router.post("/publish", summary="发布 Bot")
async def publish_bot(
    bot_id: str = Query(...),
    version: str = Query(""),
    user_uuid: str = Depends(require_login),
):
    async with CozeClient() as cli:
        data = await cli.publish_bot(bot_id, version=version)
    if data.get("code") != 0:
        return error(data.get("msg", "Coze API error"))
    return success(data.get("data"))


@router.get("/datasets/list", summary="Bot 关联知识库列表")
async def list_datasets(
    page: int = Query(1, ge=1),
    page_size: int = Query(20, ge=1, le=100),
    user_uuid: str = Depends(require_login),
):
    async with CozeClient() as cli:
        data = await cli.list_datasets(page=page, size=page_size)
    if data.get("code") != 0:
        return error(data.get("msg", "Coze API error"))
    return success(
        data.get("data", {}).get("dataset_list", []),
        total=data.get("data", {}).get("total", 0),
    )

File: client/src/api/bots.ts
/**
 * Coze Bot 管理 API
 * 对接后端: app/api/v1/bots/bots.py
 * 路由前缀: /api/v1/bots
 *
 * 后端列表响应为 {code, msg, data:[...], total},
 * 本文件统一转换为 {records, total} 以适配 useAdminTable 默认提取器。
 * 注意: 后端 create/update/delete/publish 均使用 Query 参数传值。
 */
import http from '@/utils/request'
import type { ApiResponse, PaginationResponse } from '@/types'

export interface BotListParams {
  current?: number
  size?: number
  keyword?: string
  status?: string
  spaceId?: string
  [k: string]: unknown
}

export interface Bot {
  botId: string
  name: string
  description?: string
  status?: string
  ownerId?: string
  datasetIds?: string[]
  createTime?: string | null
  updateTime?: string | null
}

// 统一构造 ApiResponse<{records, total}> 格式
function toListResult(rows: unknown[], total: number, msg = 'success'): ApiResponse<{ records: unknown[]; total: number }> {
  return {
    code: 0,
    message: msg,
    data: { records: rows, total },
    success: true,
    timestamp: Date.now(),
  } as unknown as ApiResponse<{ records: unknown[]; total: number }>
}

function toDataResult(data: unknown, msg = 'success'): ApiResponse<unknown> {
  return {
    code: 0,
    message: msg,
    data,
    success: true,
    timestamp: Date.now(),
  } as unknown as ApiResponse<unknown>
}

/** Bot 列表 */
export async function botList(params: BotListParams = {}): Promise<ApiResponse<PaginationResponse<Bot>>> {
  const res = await http.get('/api/v1/bots/list', {
    params: {
      page: params.current ?? 1,
      page_size: params.size ?? 20,
      space_id: params.spaceId || undefined,
    },
  })
  const body = (res as any).data || {}
  return toListResult(body.data || [], body.total || 0, body.msg) as unknown as ApiResponse<PaginationResponse<Bot>>
}

/** Bot 详情 */
export async function botDetail(botId: string): Promise<ApiResponse<Bot | null>> {
  const res = await http.get(`/api/v1/bots/${botId}`)
  const body = (res as any).data || {}
  return toDataResult(body.data, body.msg) as unknown as ApiResponse<Bot | null>
}

/** 创建 Bot (后端使用 Query 参数) */
export async function botCreate(payload: { name: string; description?: string; persona?: string }): Promise<ApiResponse<Bot>> {
  const res = await http.post('/api/v1/bots/create', null, {
    params: {
      name: payload.name,
      description: payload.description || '',
      persona: payload.persona || undefined,
    },
  })
  const body = (res as any).data || {}
  return toDataResult(body.data, body.msg) as unknown as ApiResponse<Bot>
}

/** 修改 Bot (后端使用 Query 参数; 后端使用 POST 而非 PUT) */
export async function botUpdate(botId: string, payload: { name?: string; description?: string; persona?: string }): Promise<ApiResponse<Bot>> {
  const res = await http.post('/api/v1/bots/update', null, {
    params: {
      bot_id: botId,
      name: payload.name || undefined,
      description: payload.description || undefined,
      persona: payload.persona || undefined,
    },
  })
  const body = (res as any).data || {}
  return toDataResult(body.data, body.msg) as unknown as ApiResponse<Bot>
}

/** 删除 Bot (后端使用 Query 参数; 后端使用 POST 而非 DELETE) */
export async function botDelete(botId: string): Promise<ApiResponse<unknown>> {
  const res = await http.post('/api/v1/bots/delete', null, {
    params: { bot_id: botId },
  })
  const body = (res as any).data || {}
  return toDataResult(body.data, body.msg)
}

/** 发布 Bot (后端使用 Query 参数) */
export async function botPublish(botId: string): Promise<ApiResponse<unknown>> {
  const res = await http.post('/api/v1/bots/publish', null, {
    params: { bot_id: botId },
  })
  const body = (res as any).data || {}
  return toDataResult(body.data, body.msg)
}

/** Bot 关联知识库列表 */
export async function botDatasetsList(params: BotListParams = {}): Promise<ApiResponse<PaginationResponse<unknown>>> {
  const res = await http.get('/api/v1/bots/datasets/list', {
    params: {
      page: params.current ?? 1,
      page_size: params.size ?? 20,
    },
  })
  const body = (res as any).data || {}
  return toListResult(body.data || [], body.total || 0, body.msg) as unknown as ApiResponse<PaginationResponse<unknown>>
}

export const botsApi = {
  botList,
  botDetail,
  botCreate,
  botUpdate,
  botDelete,
  botPublish,
  botDatasetsList,
}

export default botsApi
